docs/resources/index.md

```diff
@@ -0,0 +1,213 @@
+
+| Folder name                                                           | Contents                     |
+|-----------------------------------------------------------------------|------------------------------|
+| `advancements`                                                        | [Advancements][advancements] |
+| `damage_type`                                                         | Damage types                 |
+| `loot_tables`                                                         | [Loot tables][loottables]    |
+| `recipes`                                                             | [Recipes][recipes]           |
+| `structures`                                                          | Structures                   |
+| `tags`                                                                | [Tags][tags]                 |
+| `dimension`, `dimension_type`, `worldgen`, `neoforge/biome_modifiers` | Worldgen files               |
+| `neoforge/global_loot_modifiers`                                      | [Global loot modifiers][glm] |
+
+Additionally, they may also contain subfolders for some systems that integrate with commands. These systems are rarely used in conjunction with mods, but worth mentioning regardless:
+
+| Folder name      | Contents                       |
+|------------------|--------------------------------|
+| `chat_type`      | [Chat types][chattype]         |
+| `functions`      | [Functions][function]          |
+| `item_modifiers` | [Item modifiers][itemmodifier] |
+| `predicates`     | [Predicates][predicate]        |
+
+## `pack.mcmeta`
+
+_See also: [`pack.mcmeta` (Resource Pack)][packmcmetaresourcepack] and [`pack.mcmeta` (Data Pack)][packmcmetadatapack] on the [Minecraft Wiki][mcwiki]_
+
+`pack.mcmeta` files hold the metadata of a resource or data pack. For mods, NeoForge makes this file obsolete, as the `pack.mcmeta` is generated synthetically. In case you still need a `pack.mcmeta` file, the full specification can be found in the linked Minecraft Wiki articles.
+
+## Data Generation
+
+Data generation, colloquially known as datagen, is a way to programmatically generate JSON resource files, in order to avoid the tedious and error-prone process of writing them by hand. The name is a bit misleading, as it works for assets as well as data.
+
+Datagen is run through the Data run configuration, which is generated for you alongside the Client and Server run configurations. The data run configuration follows the [mod lifecycle][lifecycle] until after the registry events are fired. It then fires the [`GatherDataEvent`][event], in which you can register your to-be-generated objects in the form of data providers, writes said objects to disk, and ends the process.
+
+All data providers extend the `DataProvider` interface and usually require one method to be overridden. The following is a list of all data generators Minecraft and NeoForge offer (the linked articles add further information, such as helper methods):
```

Comment:
This is by far not all data providers offered by Minecraft and NeoForge, it is missing all the "report providers" (though those are rarely relevant to mods) and the `SpriteSourceProvider` and its parent, the `JsonCodecProvider`, and likely some others I missed. This should be worded more along the lines of "The following is a list of some noteworthy data generators Minecraft and NeoForge offer"